public/app/features/dashboard/panel_model.ts

```diff
@@ -51,12 +52,20 @@ const mustKeepProps: { [str: string]: boolean } = {
   events: true,
   cacheTimeout: true,
   nullPointMode: true,
+  cachedPluginOptions: true,
+};
+
+// Keep current option value when switching visualization
+const keepLatestProps: { [str: string]: boolean } = {
+  title: true,
+  description: true,
 };
 
 const defaults: any = {
   gridPos: { x: 0, y: 0, h: 3, w: 6 },
   datasource: null,
   targets: [{}],
+  cachedPluginOptions: {},
```

Comment:
is it supposed to be an object or a boolean?

Reply:
it's an object map (pluginId -> options)

Reply:
it is assigned true further up in the file?

Reply:
It looks like that in github review-mode. But its just a prop in with the same name assigned to true in the notPersistedProperties-object.

```
...
const notPersistedProperties: { [str: string]: boolean } = {
  events: true,
  fullscreen: true,
  isEditing: true,
  hasRefreshed: true,
  cachedPluginOptions: true,
};
...
```